game/requesthandler.cc

```diff
@@ -64,40 +64,129 @@ boost::asio::ip::address_v4 RequestHandler::getLocalIP(const std::string& servic
+                std::string key = kv.first;
+                //Hack to get an easy key value pair within the json object.
+                if(key == "Web interface by Niek Nooijens and Arjan Speiard, for full credits regarding Performous see /docs/Authors.txt"){
+                    key = "Credits";
+                }
+                std::replace(key.begin(), key.end(), ' ','_');
+                key = UnicodeUtil::toLower(key);;
+                jsonRoot[key] = kv.second;
+            }
+    	    return init_resp(request->create_response(),std::string("application/json"))
+        		.set_body(jsonRoot.dump())
+        		.done();
+        });
+	router->http_get(R"(/api/getCurrentPlaylist.json)", [](auto request, auto) {
+        Game* gm = Game::getSingletonPtr();
+        nlohmann::json jsonRoot = nlohmann::json::array();
+        for (auto const& song : gm->getCurrentPlayList().getList()) {
+            nlohmann::json songObject;
+            songObject["Title"] = song->title;
+            songObject["Artist"] = song->artist;
+            songObject["Edition"] = song->edition;
+            songObject["Language"] = song->language;
+            songObject["Creator"] = song->creator;
+            songObject["Duration"] = song->getDurationSeconds();
+            jsonRoot.push_back(songObject);
+        }
+        return init_resp(request->create_response(),std::string("application/json"))
+			.set_body(jsonRoot.dump())
+			.done();
+    });
+    router->http_get(R"(/api/getplaylistTimeout)", [](auto request, auto) {
+    	return init_resp(request->create_response())
+        .set_body(std::to_string(config["game/playlist_screen_timeout"].i()))
+        .done();
+    });
+	router->http_get(R"(/:path(.*))", [this](auto req, auto params){
+		std::clog << "webserver/debug: findFile returns: " << findFile(std::string(params["path"])).string() << std::endl;
+		return HandleFile(req, findFile(std::string(params["path"])).string());
+	});
+	
     router->non_matched_request_handler(
             [](auto req){
                 return req->create_response(restinio::status_not_found()).connection_close().done();
             });
 	return router;
 }
 
+restinio::request_handling_status_t RequestHandler::HandleFile(std::shared_ptr<restinio::request_t> request, std::string filePath) {
+    auto path = (!filePath.empty()) ? filePath : request->header().request_target();
+    auto fileName = path.substr(path.find_last_of("/\\") + 1);
+
+        std::string content_type = std::string();
+        if(path.find(".js") != std::string::npos) {
+            content_type = "text/javascript; charset=utf-8";
+        } else if (path.find(".css") != std::string::npos) {
+            content_type = "text/css";
+        } else if (path.find(".png") != std::string::npos) {
+            content_type = "image/png";
+        } else if (path.find(".gif") != std::string::npos) {
+            content_type = "image/gif";
+        } else if (path.find(".ico") != std::string::npos) {
```

Comment:
I would suggest
```c++
std::map<std::string, std::string> mimetypes{
    {"png", "image/png"},
    ...
};
```

Reply:
The problem with this approach is we're using dumb matching so most likely it'd end up being the same.

Reply:
Seriously I don't know what the fuck to do with you. I understand that you may not be an experienced programmer and that you could come up with less than optimal solutions. However, after seeing you work on Performous for a decade and *still* most of your commits being shit code, and when I give feedback on how to do it better, you **always** just dismiss it and do your funky style anyway. If you don't wish to listen to my feedback, I'm gonna give up. Please feel free to keep fucking up the code.

Reply:
I honestly don't know what the fuck your problem is. I do value and take your input into account but on the contrary, I'd say it's you who always does what he wants.

Further, what I'm saying is we're doing a dumb match for a ".extension", using a map would still mean iterating through every element in the map and comparing it because we don't have an extension string, properly speaking.

I'm sure we can improve the code to be more optimal but right now you're riding my ass and being utterly unsufferable over WIP code and in particular a piece of code that wasn't even written by me! (That is the code in master; I just changed the order and some of the cases).

If you want to improve something in a particular way, you can commi fixes just as much as I can.

Reply:
Let's not attack eachother please. The code was re-ordered that's all. Yes it can/should be improved but please let's be nice to eachother.